src/store/todo/todoSagas.ts

```diff
@@ -21,8 +29,62 @@ function* getTodoSagas(action: todosTypes.FetchTodos) {
   }
 }
 
+function* updateTodoTask(action: todosTypes.UpdateTodo) {
+  const { payload } = action;
+  try {
+    yield put(todosActions.updateTodoPending());
+
+    const todos: TodoType[] = yield select((state) => state.todosReducer.todos);
+
+    const index = todos.findIndex((todo) => todo.id === payload.todo.id);
+
+    todos[index] = payload.todo;
+    console.log(todos);
+    yield put(todosActions.fetchTodosResolved(todos));
+  } catch (error) {
+    yield put(todosActions.updateTodoRejected(error));
+    payload.callback(error, "ERROR");
+  }
+}
+
+function* addTodoTask(action: todosTypes.AddTodo) {
+  try {
+    yield put(todosActions.addTodoPending());
+    const todos: TodoType[] = yield select((state) => state.todosReducer.todos);
+    const newTodos = [...todos];
+    newTodos.push(action.payload.todo);
+    yield put(todosActions.addTodoResolved(newTodos));
+  } catch (error) {
+    yield put(todosActions.addTodoRejected(error));
+  }
+}
+
+function* deleteTodoTask(action: todosTypes.DeleteTodo) {
+  try {
+    yield put(todosActions.deleteTodoPending());
+    const todos: TodoType[] = yield select((state) => state.todosReducer.todos);
```

Comment:
Like I siad, use your delay method here to simulate asynch, then user will be able to see loading indicator.